[15.0][MIG] partner_pricelist_search: Migration to 15.0

victoralmau opened the pull request:
Superseed: https://github.com/OCA/partner-contact/pull/1254

Migration to 15.0

Please @pedrobaeza and @carlosdauden can you review it?

@Tecnativa TT37297

carlosdauden committed:
[ADD] partner_pricelist_search: New module
TT27659

carlosdauden committed:
[MIG] partner_pricelist_search: Migration to v13.0
TT28442

pedrobaeza committed:
[IMP] partner_pricelist_search: black, isort, prettier

victoralmau committed:
[MIG] partner_pricelist_search: Migration to 14.0

[UPD] Update partner_pricelist_search.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-14.0/partner-contact-14.0-partner_pricelist_search
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_pricelist_search/

sebalix committed:
[FIX] Fix access right issues when using actions

partner_pricelist_search 14.0.1.0.1

[IMP] update dotfiles [ci skip]

vince-dynapps committed:
[MIG] partner_pricelist_search: Migration to 15.0

TT37297

rousseldenis commented:
/ocabot migration partner_pricelist_search

rousseldenis commented on partner_pricelist_search/models/product_pricelist.py:
```diff
@@ -0,0 +1,21 @@
+            domain = [("property_product_pricelist", "=", self.id)]
+        else:
+            domain = [("property_product_pricelist", "in", self.ids)]
+        partners = self.env["res.partner"].search(domain)
```
You can put the domain directly in action to avoid a search

rousseldenis commented on partner_pricelist_search/models/res_partner.py:
```diff
@@ -0,0 +1,43 @@
+                % operator
+            )
+        partners = self.with_context(prefetch_fields=False).search([])
+        return [("id", "in", partners.filtered(filter_func).ids)]
```
@victoralmau Maybe you can enhance this and use filtered_domain([("property_product_pricelist", operator, value)]) directy and remove all the filter_func?

chienandalu committed:
[FIX] partner_pricelist_search: speedup search

TT39742

victoralmau commented:
Changes done. Please @carlosdauden could you review it again?

pedrobaeza commented:
We are doing some improvements in v13, so better to wait to include it.

carlosdauden reviewed (approved):
Fantastic, I had pending to make those changes.

sergio-teruel committed:
[FIX] partner_pricelist_search: Remove list type also from search arguments

[UPD] Update partner_pricelist_search.pot

partner_pricelist_search 14.0.1.0.2

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-14.0/partner-contact-14.0-partner_pricelist_search
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_pricelist_search/

victoralmau committed:
[IMP] partner_pricelist_search: black, isort, prettier

victoralmau commented:
Added https://github.com/OCA/partner-contact/pull/1384 to commit history.

pedrobaeza commented:
@rousseldenis can you please re-review you "requested changes"?

pedrobaeza commented:
/ocabot merge nobump
